src/main/scala/io/citrine/lolo/bags/BaggedResult.scala

```diff
@@ -434,11 +469,155 @@ case class MultiTaskBaggedResult(baggedPredictions: Seq[BaggedResult[Any]], real
+        val numPredictions = baggedPredictionsI.size
+        // mean value for each prediction
+        val expectedI = baggedPredictionsI.map(ps => utils.mean(ps))
+        val expectedJ = baggedPredictionsJ.map(ps => utils.mean(ps))
+        // Stick the individual predictions into Breeze matrices
+        val predMatI = new DenseMatrix[Double](numBags, numPredictions, baggedPredictionsI.flatten.toArray)
+        val predMatJ = new DenseMatrix[Double](numBags, numPredictions, baggedPredictionsJ.flatten.toArray)
+        // Perform the jackknife calculation on the predictions for labels i and j, then multiply them together
+        // with factor of (n - 1) / n, where n is the number of training data
+        val JMatI = NibJMat.t * predMatI
+        val JMatJ = NibJMat.t * predMatJ
+        val JMat2: DenseMatrix[Double] = JMatI *:* JMatJ * ((Nib.size - 1.0) / Nib.size)
+        // Perform the infinitesimal jackknife calculation on the predictions for labels i and j,then multiply them together
+        val IJMatI = NibIJMat.t * predMatI
+        val IJMatJ = NibIJMat.t * predMatJ
+        val IJMat2: DenseMatrix[Double] = IJMatI *:* IJMatJ
+        // Add J(ackknife) and IJ covariance terms together
+        val totalCovarianceTerm = JMat2 + IJMat2
+        // Calculate 1/B^2 once, to avoid doing it in the loop
+        val inverseSize2 = math.pow(1.0 / numBags, 2.0)
+        // Loop over predictions, and for each one calculate the bias correction term and subtract it from each covariance term
+        // The resulting structure has size (# predictions) x (# bags)
+        val scores = baggedPredictionsI.indices.map { k =>
+          val correctionI = predMatI(::, k) - expectedI(k)
+          val correctionJ = predMatJ(::, k) - expectedJ(k)
+          val correction = correctionI.dot(correctionJ) * inverseSize2
+          0.5 * (totalCovarianceTerm(::, k) - math.E * correction)
+        }.map(_.toScalaVector())
+        // For each prediction, rectify the covariance scores to compute correlation
+        Some((scores, sigmaISeq, sigmaJSeq).zipped.map { (trainingContributions, sigmaI, sigmaJ) =>
+          BaggedResult.rectifyCorrelationScores(trainingContributions, sigmaI, sigmaJ)
+        })
+      case _: Any => None
+    }
+  }
+
+  private def getUncertaintyCorrelationJackknifeExplicit(i: Int, j: Int): Option[Seq[Double]] = {
+    val sigmaIOption = baggedPredictions(i).getUncertainty(observational = false).map(_.asInstanceOf[Seq[Double]])
+    val sigmaJOption = baggedPredictions(j).getUncertainty(observational = false).map(_.asInstanceOf[Seq[Double]])
+    (sigmaIOption, sigmaJOption) match {
+      case (Some(sigmaISeq), Some(sigmaJSeq)) =>
+        Some(Seq.tabulate(numPredictions) { predIndex => // loop over each prediction input
+          // Get the individual tree predictions for this input
+          val treePredictionsI = baggedPredictions(i).predictions.map(_.getExpected()(predIndex)).asInstanceOf[Seq[Double]]
```

Comment:
If this explicit method is kept in the final version and performance is a concern, moving these lines outside the loop and then indexing them for each `predIndex` is an easy optimization.